File: anttask/src/main/java/org/dyndns/fichtner/purgeannotationrefs/ant/types/AnnotationRemoveSelector.java
package org.dyndns.fichtner.purgeannotationrefs.ant.types;

import org.dyndns.fichtner.purgeannotationrefs.Matcher.RegExpMatcher;
import org.dyndns.fichtner.purgeannotationrefs.Matcher.StringMatcher;
import org.dyndns.fichtner.purgeannotationrefs.RemoveFrom;

import java.util.function.Predicate;
import java.util.regex.Pattern;

public class AnnotationRemoveSelector {

  private RemoveFrom from;
  private Predicate<String> matcher;

  public AnnotationRemoveSelector() {
    super();
  }

  public RemoveFrom getFrom() {
    return this.from;
  }

  public void setFrom(RemoveFrom removeFrom) {
    this.from = removeFrom;
  }

  public Predicate<String> getMatcher() {
    return this.matcher;
  }

  public void setName(String annoName) {
    this.matcher = new StringMatcher(annoName);
  }

  public void setRegexp(String regexp) {
    this.matcher = new RegExpMatcher(Pattern.compile(regexp));
  }

}

File: gradle-plugin/src/main/java/org/dyndns/fichtner/purgeannotationrefs/gradle/AnnotationRemoveSelector.kt
package org.dyndns.fichtner.purgeannotationrefs.gradle

import org.dyndns.fichtner.purgeannotationrefs.Matcher.RegExpMatcher
import org.dyndns.fichtner.purgeannotationrefs.Matcher.StringMatcher
import org.dyndns.fichtner.purgeannotationrefs.RemoveFrom
import java.util.function.Predicate
import java.util.regex.Pattern

data class AnnotationRemoveSelector(var from: RemoveFrom = RemoveFrom.ALL, var matcher: Predicate<String>) {
  companion object {
    @JvmStatic
    @JvmOverloads
    fun named(annotationName: String, from: RemoveFrom = RemoveFrom.ALL): AnnotationRemoveSelector {
      return AnnotationRemoveSelector(from, StringMatcher(annotationName))
    }

    @JvmStatic
    @JvmOverloads
    fun regexp(regexp: String, from: RemoveFrom = RemoveFrom.ALL): AnnotationRemoveSelector {
      return AnnotationRemoveSelector(from, RegExpMatcher(Pattern.compile(regexp)))
    }

    @JvmStatic
    @JvmOverloads
    fun matching(from: RemoveFrom = RemoveFrom.ALL, matcher: Predicate<String>): AnnotationRemoveSelector {
      return AnnotationRemoveSelector(from, matcher)
    }
  }
}